src/detect-engine.c

```diff
@@ -2025,26 +2032,6 @@ static int DetectEnginePktInspectionAppend(Signature *s, InspectionBufferPktInsp
     return 0;
 }
 
-int DetectEnginePktInspectionSetup(Signature *s)
-{
-    /* only handle PMATCH here if we're not an app inspect rule */
-    if (s->sm_arrays[DETECT_SM_LIST_PMATCH] && (s->init_data->init_flags & SIG_FLAG_INIT_STATE_MATCH) == 0) {
```

Comment:
there is a check for this SIG_FLAG_INIT_STATE_MATCH flag here, where did that go?